workspaces/libnpmexec/lib/index.js

```diff
@@ -86,20 +85,47 @@ const exec = async (opts) => {
     packages.push(args[0])
   }
 
+  // figure out whether we need to install stuff, or if local is fine
+  const localArb = new Arborist({
+    ...flatOptions,
+    path,
+  })
+  const localTree = await localArb.loadActual()
+
+  const getLocalManifest = ({ tree, name }) => {
+    // look up the package name in the current tree inventory,
+    // if it's found then return that normalized pkg data
+    let node
+    const nodes = tree.inventory.query('packageName', name)
+    for (const n of nodes) {
+      node = n
+      break
```

Comment:
this is because the above `tree.inventory.query()` returns a `Set` and we want the first item? if so, might be worth a comment because otherwise this pattern feels weird at a glance.

i've done `nodes().values().next().value` in the past, but i don't think that's really any more readable lol

Reply:
if you do the `nodes.size` check first, then you can do `const [node] = nodes` and _that_ feels better

Reply:
actually.. you can skip the size check entirely now that i think about it.

```
const [node] = tree.inventory.query()
return node
```

is plenty

Reply:
omg that's awesome 😍 thanks @nlf